services/web/server/src/simcore_service_webserver/projects/_trash_api.py

```diff
@@ -0,0 +1,59 @@
+import asyncio
+import logging
+from datetime import timedelta
+
+import arrow
+from aiohttp import web
+from models_library.products import ProductName
+from models_library.projects import ProjectID
+from models_library.users import UserID
+
+from . import projects_api
+from .models import ProjectPatchExtended
+from .settings import get_plugin_settings
+
+_logger = logging.getLogger(__name__)
+
+
+async def empty_trash(app: web.Application, product_name: ProductName, user_id: UserID):
+    # filter trashed=True and set them to False
+    _logger.debug(
+        "CODE PLACEHOLDER: all projects marked of trashed of %s in %s are deleted",
+        f"{user_id=}",
+        f"{product_name=}",
+    )
+    raise NotImplementedError
+
+
+async def prune_all_trashes(app: web.Application) -> list[str]:
+    settings = get_plugin_settings(app)
+    retention = timedelta(days=settings.PROJECTS_TRASH_RETENTION_DAYS)
+
+    _logger.debug(
+        "CODE PLACEHOLDER: **ALL** projects marked as trashed during %s days are deleted",
+        retention,
+    )
+    await asyncio.sleep(5)
+
+    return []
+
+
+async def update_project(
+    app: web.Application,
+    *,
+    product_name: ProductName,
+    user_id: UserID,
+    project_id: ProjectID,
+    trashed: bool,
+):
+    # FIXME: can you trash something that is running?
```

Comment:
well currently you can delete something that is running... so I guess we need to define what to do

Reply:
thx! Yes, I forgot thisone. I have added the missing code.

My goal is in the following PRs to unify and compact the APIs of `delete` and `trash`.

Reply:
Actually, the logic is already wrong if we compare with e.g. docker  api. 
If a resource is "active" (e.g. running, mounted etc) , `delete` will fail 

![image](https://github.com/user-attachments/assets/2942465a-1b9f-4639-bcf2-5b0c22ef6284)


Except with `--force`

Reply:
I added an extra error handling layer . It is a (good) start.

Reply:
we have force in the dv-2 by the way...